### PR TITLE
Fix catalog lookup

### PR DESCRIPTION
**Pull request by moliholy**
Fixes a problem when retrieving a catalog given a path

### CODE REVIEW

File: cvmfs/catalog.py [hunk 1]
```diff
@@ -209,10 +209,14 @@ def find_nested_for_path(self, needle_path):
         nested_catalogs  = self.list_nested()
         best_match       = None
         best_match_score = 0
+        num_slashes      = needle_path.count('/')
```
Comment (reneme) [inline cvmfs/catalog.py]: probably you want to retrieve this information from `real_needle_path` two lines below.

File: cvmfs/catalog.py [hunk 2]
```diff
@@ -209,10 +209,14 @@ def find_nested_for_path(self, needle_path):
...
         real_needle_path = self._canonicalize_path(needle_path)
         for nested_catalog in nested_catalogs:
             if real_needle_path.startswith(nested_catalog.root_path) and \
-                len(nested_catalog.root_path) > best_match_score:
+               len(nested_catalog.root_path) > best_match_score:
+                    if num_slashes == nested_catalog.root_path.count('/') and \
+                       needle_path != nested_catalog.root_path:
```
Comment (reneme) [inline cvmfs/catalog.py]: Please combine these statements into a private predicate function like so (please double-check if that actually fits)?

``` python
def __is_better_match(needle_path, candidate_path, current_best_score):
  return needle_path.startswith(candidate_path)               and \
         len(candidate_path) > current_best_score             and \
         (needle_path.count("/") != candidate_path.count('/') or  \
          needle_path == candidate_path)  
```

**Comment by moliholy** [comment]
Please don't approve this PR by now, I have just found another bug related with this one

**Comment by moliholy** [comment]
OK, now it works. There was still an error when processing longer paths

**Pull request merged**